### ISSUE TITLE
Change `declaration-block-no-duplicate-properties` option to `consecutive-duplicates-with-different-syntaxes`

### ISSUE THREAD

**Issue by ybiquitous** [stylelint/stylelint-config-recommended#186]
### What is the problem you're trying to solve?

Currently, the rule has an option of `ignore: ['consecutive-duplicates-with-different-values']`:

https://github.com/stylelint/stylelint-config-recommended/blob/e6c852041a7c030c5763b3eca0deddc9fad852a9/index.js#L12-L17

### What solution would you like to see?

The next version of Stylelint will have a new option `ignore: ['consecutive-duplicates-with-different-syntaxes']`. If changing the current option to a new one, we can detect the following problematic code:

```css
a {
  width: min-content;
  width: max-content; /* Currently, the rule doesn't warn due to `consecutive-duplicates-with-different-values` */
}
```

Ref: https://github.com/stylelint/stylelint/issues/6632

Note: This change will produce a major update because detected problems will increase.

**Comment by kimulaco** [comment]
I'll work on this issue if no problem.

**Comment by ybiquitous** [comment]
@kimulaco Thanks for the proposal. We're glad about your contribution!

But please note that we cannot merge a pull request in this repo now, since it's necessary to wait for the next version of Stylelint.

We will ship `stylelint`, `stylelint-config-recommended`, and `stylelint-config-standard` in the next release flow. See also our [maintainer guide](https://stylelint.io/maintainer-guide/releases).
